test/fuzzy_object_store/fuzz_object.cpp

```diff
@@ -0,0 +1,569 @@
+{
+    log << "Open/close realm\n";
+    shared_realm->close();
+    shared_realm.reset();
+    shared_realm = Realm::get_shared_realm(config);
+    log << "Verify group after realm got reopened\n";
+    auto& group = shared_realm->read_group();
+    REALM_DO_IF_VERIFY(log, group.verify());
+}
+
+void FuzzObject::create_table_view(Group& group, FuzzLog& log, State& s, std::vector<TableView>& table_views)
+{
+    log << "FuzzObject::create_table_view();\n";
+    TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+    TableRef t = group.get_table(table_key);
+    log << "table_views.push_back(wt->get_table(" << table_key << ")->where().find_all());\n";
+    TableView tv = t->where().find_all();
+    table_views.push_back(tv);
+}
+
+void FuzzObject::check_null(Group& group, FuzzLog& log, State& s)
+{
+    log << "FuzzObject::check_null();\n";
+    TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+    TableRef t = group.get_table(table_key);
+    if (t->get_column_count() > 0 && t->size() > 0) {
+        auto all_col_keys = t->get_column_keys();
+        size_t ndx = get_next_token(s) % all_col_keys.size();
+        ColKey col = all_col_keys[ndx];
+        ObjKey key = t->get_object(get_int32(s) % t->size()).get_key();
+        log << "group.get_table(" << table_key << ")->get_object(" << key << ").is_null(" << col << ");\n";
+        bool res = t->get_object(key).is_null(col);
+        static_cast<void>(res);
+    }
+}
+
+void FuzzObject::async_write(SharedRealm shared_realm, FuzzLog& log)
+{
+    log << "FuzzObject::async_write\n";
+    if (!shared_realm->is_in_async_transaction() && !shared_realm->is_in_transaction()) {
+        shared_realm->async_begin_transaction([&]() {
+            shared_realm->async_commit_transaction([](std::exception_ptr) {});
+        });
+    }
+}
+
+void FuzzObject::async_cancel(SharedRealm shared_realm, Group& group, FuzzLog& log, State& s)
+{
+    log << "FuzzObject::async_cancel\n";
+    if (!shared_realm->is_in_async_transaction() && !shared_realm->is_in_transaction()) {
+        auto token = shared_realm->async_begin_transaction([&]() {
+            TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+            size_t num_rows = get_next_token(s);
+            if (group.get_table(table_key)->size() + num_rows < max_rows) {
+                log << "{ std::vector<ObjKey> keys; wt->get_table(" << table_key << ")->create_objects("
+                    << num_rows % add_empty_row_max << ", keys); }\n";
+                std::vector<ObjKey> keys;
+                group.get_table(table_key)->create_objects(num_rows % add_empty_row_max, keys);
+            }
+        });
+        shared_realm->async_cancel_transaction(token);
+    }
+}
+
+DataType FuzzObject::get_type(unsigned char c) const
+{
+    DataType types[] = {type_Int, type_Bool, type_Float, type_Double, type_String, type_Binary, type_Timestamp};
```

Comment:
Missing newer types: UUID, ObjectID, Decimal128 + sets/dictionaries. But I suppose that can come later?

Reply:
Add TODO

Reply:
Yes, it will come later. I propose to logging all this into issues that I can start implementing as soon as this goes live.